### PR TITLE
Marcos/test pr 6054

### PR DESCRIPTION
**Pull request by marcosmarxm**
## What
Test Community PR #6054

## How
*Describe the solution*

## Recommended reading order
1. `x.java`
2. `y.python`

## Pre-merge Checklist
Expand the relevant checklist and delete the others. 

<details><summary> <strong> New Connector </strong></summary>
<p>
   
#### Community member or Airbyter
   
- [ ] **Community member?** Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret` 
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated 
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] `docs/SUMMARY.md`
    - [ ] `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
    - [ ] `docs/integrations/README.md`
    - [ ] `airbyte-integrations/builds.md`
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
- [ ] Connector added to connector index like described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
   
#### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items. 
   
- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] Credentials added to Github CI. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci). 
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing. 
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)
   
</p>
</details>

<details><summary> <strong> Updating a connector </strong></summary>
<p>
   
#### Community member or Airbyter
   
- [ ] Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret` 
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated 
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] Changelog updated in `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
- [ ] Connector version bumped like described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
   
#### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items. 
   
- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] Credentials added to Github CI. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci). 
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing. 
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)

</p>
</details>

<details><summary> <strong> Connector Generator </strong> </summary>
<p>
   
- [ ] Issue acceptance criteria met
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
- [ ] If adding a new generator, add it to the [list of scaffold modules being tested](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connector-templates/generator/build.gradle#L41)
- [ ] The generator test modules (all connectors with `-scaffold` in their name) have been updated with the latest scaffold by running `./gradlew :airbyte-integrations:connector-templates:generator:testScaffoldTemplates` then checking in your changes
- [ ] Documentation which references the generator is updated as needed.
</p>
</details>

### CODE REVIEW

File: airbyte-integrations/connectors/source-facebook-marketing/source_facebook_marketing/source.py [hunk 1]
```diff
@@ -42,6 +43,13 @@ class Config:
         examples=["2017-01-25T00:00:00Z"],
     )
 
+    end_date: Optional[datetime] = Field(
+        description="The end date until which you'd like to replicate data for AdCreatives and AdInsights APIs, in the format YYYY-MM-DDT00:00:00Z. All data generated between start_date and this date will be replicated.",
+        pattern="^[0-9]{4}-[0-9]{2}-[0-9]{2}T[0-9]{2}:[0-9]{2}:[0-9]{2}Z$",
+        examples=["2017-01-26T00:00:00Z"],
+        default=None,
+    )
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-facebook-marketing/source_facebook_marketing/source.py:51]: @keu The change was to made this Field as Optional and use default None, after set end_date to now.

Comment (keu) [reply airbyte-integrations/connectors/source-facebook-marketing/source_facebook_marketing/source.py:51]: https://github.com/samuelcolvin/pydantic/pull/1210 please use inst

File: airbyte-integrations/connectors/source-facebook-marketing/source_facebook_marketing/source.py [hunk 2]
```diff
@@ -87,17 +95,21 @@ def streams(self, config: Mapping[str, Any]) -> List[Type[Stream]]:
         """
         config: ConnectorConfig = ConnectorConfig.parse_obj(config)  # FIXME: this will be not need after we fix CDK
         api = API(account_id=config.account_id, access_token=config.access_token)
+        if not config.end_date:
+            config.end_date = pendulum.now()
+
```
Comment (keu) [inline airbyte-integrations/connectors/source-facebook-marketing/source_facebook_marketing/source.py:100]:
```suggestion
```

File: airbyte-integrations/connectors/source-facebook-marketing/unit_tests/test_client.py [hunk 3]
```diff
@@ -91,7 +91,7 @@ def test_limit_reached(self, requests_mock, api, fb_call_rate_response, account_
         requests_mock.register_uri("GET", FacebookSession.GRAPH + f"/{FB_API_VERSION}/1/", [{"status_code": 200}])
         requests_mock.register_uri("GET", FacebookSession.GRAPH + f"/{FB_API_VERSION}/2/", [{"status_code": 200}])
 
-        stream = Campaigns(api=api, start_date=datetime.now(), include_deleted=False)
+        stream = Campaigns(api=api, start_date=datetime.now(), end_date=datetime.now(), include_deleted=False)
```
Comment (keu) [inline airbyte-integrations/connectors/source-facebook-marketing/unit_tests/test_client.py:94]: for consistency it is better to use pendulum.now

**Review by keu — CHANGES_REQUESTED** [review]
LGTM

File: airbyte-integrations/connectors/source-facebook-marketing/source_facebook_marketing/source.py (hunk 2 above)
Comment (marcosmarxm) [reply airbyte-integrations/connectors/source-facebook-marketing/source_facebook_marketing/source.py:100]: @keu the default value is None, if we remove this part we're going to send None to pendulum.instance when creating the object. WDYT?

Comment (keu) [reply airbyte-integrations/connectors/source-facebook-marketing/source_facebook_marketing/source.py:100]: not if you use default_factory

File: airbyte-integrations/connectors/source-facebook-marketing/source_facebook_marketing/source.py (hunk 1 above)
Comment (marcosmarxm) [reply airbyte-integrations/connectors/source-facebook-marketing/source_facebook_marketing/source.py:51]: having problem adding `default_factory=pendulum.now` can you help here @keu ?

Comment (keu) [reply airbyte-integrations/connectors/source-facebook-marketing/source_facebook_marketing/source.py:51]:
```
iff --git a/airbyte-integrations/connectors/source-facebook-marketing/source_facebook_marketing/source.py b/airbyte-integrations/connectors/source-facebook-marketing/source_facebook_marketing/source.py
index 386446175..b9a8857c8 100644
--- a/airbyte-integrations/connectors/source-facebook-marketing/source_facebook_marketing/source.py
+++ b/airbyte-integrations/connectors/source-facebook-marketing/source_facebook_marketing/source.py
@@ -47,7 +47,7 @@ class ConnectorConfig(BaseModel):
         description="The end date until which you'd like to replicate data for AdCreatives and AdInsights APIs, in the format YYYY-MM-DDT00:00:00Z. All data generated between start_date and this date will be replicated.",
         pattern="^[0-9]{4}-[0-9]{2}-[0-9]{2}T[0-9]{2}:[0-9]{2}:[0-9]{2}Z$",
         examples=["2017-01-26T00:00:00Z"],
-        default=None,
+        default_factory=pendulum.now,
     )

     include_deleted: bool = Field(default=False, description="Include data from deleted campaigns, ads, and adsets.")
```
works for me
```
start_date=datetime.datetime(2021, 5, 23, 0, 0, tzinfo=datetime.timezone.utc) 
end_date=DateTime(2021, 10, 5, 17, 11, 13, 63974, tzinfo=Timezone('Etc/UTC'))
```

**Review by sherifnada — APPROVED** [review]
i reveiewed the other PR with some suggestions, could you make sure to pick those up?

**Pull request merged**